resources/lib/widgets.py

```diff
@@ -378,6 +379,11 @@ def get_widget_content(handle, params):
         url_params["userid"] = user_id
         url_params["Recursive"] = True
         url_params["ImageTypeLimit"] = 1
+        rewatch_days = int(settings.getSetting("rewatch_days"))
+        if rewatch_days > 0:
+            rewatch_since = datetime.datetime.today() - datetime.timedelta(days=rewatch_days)
+            url_params["nextUpDateCutoff"] = rewatch_since.strftime("%Y-%m-%d")
```

Comment:
I haven't had a chance to test yet, but this changes the entirety of next up, not just the rewatching feature, no?  That doesn't seem like an intentional change that should be included here

Reply:
This would indeed change the complete behavior of next up, but it would be the same behavior as the current android client.
Additionally it would change it "only" if the rewatch feature is enabled in the addon settings. 
The problem is, that without the date cutoff, the query returns a bunch of episodes that are not relevant (rewatch finished months ago or stopped).
I see two alternatives to the current implementation:

1. Make it a new separate widget
2. Make a second API call and combine data of next up with next up + rewatch and filter out duplicates

Reply:
I don't have too much of an opinion, honestly.  I personally don't see the need for the feature, and from what I saw after it was added it's pretty finnicky (or was, at least).  I'll likely never use it myself.

Doing multiple api calls will slow the menu down quite a bit.  There's the second network round trip of actually doing the request, then merging the data locally.  It wouldn't surprise me if it comes down to anywhere from half a second to a full second slower.  But it does eliminate the changing behavior of non-rewatching episodes not all being displayed.

Basically, ¯\\\_(ツ)_/¯.  I would say implement it whichever way you think is best, and if it's problematic we can always adjust it later in a future release.  As long as it's disabled by default (which it currently is).